test/ApiNodeService.test.ts

```diff
@@ -110,4 +112,25 @@ describe('ApiNodeService', () => {
 		expect(apiStatus.restVersion).to.be.equal(serverInfo.restVersion);
 		expect(apiStatus.chainHeight).to.be.equal(chainInfo.height);
 	}).timeout(10_000);
+
+	it('closes websocket connection after checking websocket health', async () => {
+		// Arrange:
+		const close = stub();
+		const clientWsMock = ({
+			on: (event: string, callback: any) => {
+				if (event === 'open' || event === 'error') {
+					callback();
+				}
+			},
+			close,
+		} as unknown) as WebSocket;
+
+		stub(ApiNodeService, 'createWebSocketClient').returns(clientWsMock);
+
+		// Act:
+		await ApiNodeService.checkWebSocketHealth(host, 3000, 'wss');
+
+		// Assert:
+		expect(close.calledTwice).to.be.eq(true);
```

Comment:
why calledTwice?

Reply:
It's because of the testing logic; `clientWsMock.on` method calls the `open` and `error` callback handlers right away, that's why we expect, 2 calls, 1 for `open` and 1 for the `error` callbacks.